tools/src/main/kotlin/ktx/tools/BundleLinesCreator.kt

```diff
@@ -0,0 +1,257 @@
+import org.apache.commons.io.filefilter.TrueFileFilter
+
+/**
+ * Searches for properties files and generates Ktx BundleLine class files for them when executed. The companion object
+ * instance can be used directly or behavior can be customized by subclassing.
+ * */
+open class BundleLinesCreator {
+
+  companion object Default : BundleLinesCreator()
+
+  internal fun execute(params: BundleLinesCreatorParams) {
+    with(params) {
+      execute(
+        targetPackage = requireTargetPackage(),
+        explicitBundlesDirectory = bundlesDirectory,
+        searchSubdirectories = searchSubDirectories,
+        targetSourceDirectory = targetSourceDirectory,
+        enumClassName = enumClassName,
+        codeIndent = codeIndent
+      )
+    }
+  }
+
+  /**
+   * Executes creation of enum source files.
+   * @param targetPackage The package of the generated enums.
+   * @param explicitBundlesDirectory Path directory that is searched for properties file, relative to [pathPrefix]. If
+   * null (the default), a directory is searched for as described by [findFallbackAssetDirectory].
+   * @param searchSubdirectories Whether to search subdirectories of the parent directory. Default true.
+   * @param targetSourceDirectory The directory enum source files will be placed in, relative to [pathPrefix]. Default
+   * `"core/src"`.
+   * @param enumClassName The name of the generated enum class. If non-null, a single enum class is created for all
+   * bundles found. If null, each unique base bundle's name will be used for a distinct enum class. Default `"Nls"`.
+   * @param codeIndent The whitespace String used as an indent in the generated code. Default four space characters.
+   * @param pathPrefix A prefix applied for all file operations. Default is empty, so the working directory is the
+   * parent.
+   */
+  fun execute(
+    targetPackage: String,
+    explicitBundlesDirectory: String? = null,
+    searchSubdirectories: Boolean = true,
+    targetSourceDirectory: String = "core/src",
+    enumClassName: String? = "Nls",
+    codeIndent: String = "    ",
+    pathPrefix: String = ""
+  ) {
+    require(codeIndent.all(Char::isWhitespace)) { "codeIntent must consist entirely of whitespace." }
+    val parentDirectory = explicitBundlesDirectory?.let { File(pathPrefix + it) }
+      ?: findFallbackAssetDirectory(pathPrefix)
+    requireNotNull(parentDirectory) { "Failed to find asset directory." }
+    val baseFiles = findBasePropertiesFiles(parentDirectory, searchSubdirectories)
+    val enumNamesToEntryNames = collectEnums(baseFiles, enumClassName)
+    val outDir = File("$pathPrefix$targetSourceDirectory/${targetPackage.replace(".", "/")}")
+      .apply { mkdirs() }
+    for ((enumName, entryNames) in enumNamesToEntryNames) {
+      val outFile = File(outDir, "$enumName.kt")
+      val sourceCode = generateKtFileContent(targetPackage, enumName, entryNames, codeIndent)
+      outFile.writeText(sourceCode)
+    }
+    println(
```

Comment:
I don't think this is the correct way of logging, we should rely on Gradle logger instead so it honors the logging level settings.

https://docs.gradle.org/current/userguide/logging.html